adapters/criteo/models.go

```diff
@@ -0,0 +1,334 @@
+package criteo
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type criteoRequest struct {
+	ID          *string              `json:"id,omitempty"`
+	Publisher   *criteoPublisher     `json:"publisher,omitempty"`
+	User        *criteoUser          `json:"user,omitempty"`
+	GdprConsent *criteoGdprConsent   `json:"gdprconsent,omitempty"`
+	Slots       []*criteoRequestSlot `json:"slots,omitempty"`
+}
```

Comment:
Why do all of the models using pointer fields?

Reply:
You are right, I just removed those useless pointers :)